docs/ref/models/expressions.txt

```diff
@@ -615,6 +613,12 @@ This is the only way to perform an aggregation within a ``Subquery``, as
 using :meth:`~.QuerySet.aggregate` attempts to evaluate the queryset (and if
 there is an ``OuterRef``, this will not be possible to resolve).
 
+Using Expressions to filter QuerySets
+-------------------------------------
+
+When an Expression outputs a boolean value, it may be used to directly
+filter a queryset, or as the argument to a ``When`` expression.
```

Comment:
I think the `When` thing needs documenting on `Case` / `When` rather than under a heading here. I'm also dubious that we need a separate heading for this, it could just be a sentence in the intro paragraph, plus some entries in the big list of examples at the top?

Reply:
Oh, this change slipped under the radar after other changes.